### ISSUE TITLE
v17.0 builds but is not listed in pgenv versions

### ISSUE THREAD

**Issue by sathishmanohar** [theory/pgenv#81]
```sh
pgenv use 17.0
PostgreSQL 17.0 not installed; installed versions:
  *   11.10     pgsql-11.10
      16.4      pgsql-16.4
```

```sh
pgenv versions
  *   11.10     pgsql-11.10
      16.4      pgsql-16.4
```

```sh
ls .pgenv
bin  config  patch  pgsql  pgsql-11.10  pgsql-16.4  src  LICENSE.md  README.md
```

```sh
ls ~/.pgenv/src
postgresql-11.10  postgresql-16.4  postgresql-17.0  postgresql-11.10.tar.bz2  postgresql-16.4.tar.bz2  postgresql-17.0.tar.bz2
```

I'm on Arch Linux BTW

**Comment by fluca1978** [comment]
Seems it is not installed at all, since you don't have `pgsql-17.0` in your `PGENV_ROOT`.
However, it does list 17 when installed:

```
% pgenv versions
      16.0      pgsql-16.0
      16.1      pgsql-16.1
      16.2      pgsql-16.2
  *   17.0      pgsql-17.0
```

**Comment by fljdin** [comment]
I'm facing the same issue with 17.0

```console
$ pgenv build 17.0
...
make[1]: Leaving directory '/var/lib/pgenv/src/postgresql-17.0/src'

$ pgenv versions
Using PGENV_ROOT /var/lib/pgenv
      15.4      pgsql-15.4
  *   15.6      pgsql-15.6
      16.0      pgsql-16.0
```

**Comment by fluca1978** [comment]
> I'm facing the same issue with 17.0
> 
> ```
> $ pgenv build 17.0
> ...
> make[1]: Leaving directory '/var/lib/pgenv/src/postgresql-17.0/src'
> 


Are there errors reported in the output?
What version of `pgenv` are you running?

**Comment by theory** [comment]
Might the build be failing due to #79?

**Comment by sathishmanohar** [comment]
```
make[3]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/contrib/spi'
cp ../../../contrib/spi/refint.so refint.so
cp ../../../contrib/spi/autoinc.so autoinc.so
make[2]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/test/regress'
/usr/bin/make -C test/isolation all
make[2]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/test/isolation'
/usr/bin/make -C ../../../src/interfaces/libpq all
/usr/bin/make -C ../../../src/port all
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 -I. -I. -I../../../src/interfaces/libpq -I./../regress -I../../../src/include  -D_GNU_SOURCE   -c -o isolationtester.o isolationtester.c
make[3]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/interfaces/libpq'
/usr/bin/make -C ../../../src/port all
make[3]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
/usr/bin/make -C ../../../src/common all
make[4]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
/usr/bin/make -C ../../../src/common all
make[3]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
make[3]: Nothing to be done for 'all'.
make[3]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
/usr/bin/bison -Wno-deprecated  -d -o specparse.c specparse.y
make[4]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
make[3]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/interfaces/libpq'
/usr/bin/flex   -o'specscanner.c' specscanner.l
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 -I. -I. -I../../../src/interfaces/libpq -I./../regress -I../../../src/include  -D_GNU_SOURCE   -c -o isolation_main.o isolation_main.c
/usr/bin/make -C ../../../src/test/regress pg_regress.o
make[3]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/test/regress'
/usr/bin/make -C ../../../src/port all
make[4]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/port'
/usr/bin/make -C ../../../src/common all
make[4]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
make[4]: Nothing to be done for 'all'.
make[4]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/common'
make[3]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/test/regress'
touch specparse.h
rm -f pg_regress.o && ln -s ../../../src/test/regress/pg_regress.o .
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 -I. -I. -I../../../src/interfaces/libpq -I./../regress -I../../../src/include  -D_GNU_SOURCE   -c -o specparse.o specparse.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 -I. -I. -I../../../src/interfaces/libpq -I./../regress -I../../../src/include  -D_GNU_SOURCE   -c -o specscanner.o specscanner.c
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 isolation_main.o pg_regress.o -L../../../src/interfaces/libpq -lpq -L../../../src/port -L../../../src/common   -Wl,--as-needed -Wl,-rpath,'/home/user/.pgenv/pgsql-17.0/lib',--enable-new-dtags  -lpgcommon -lpgport -lz -lreadline -lm  -o pg_isolation_regress
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Wdeclaration-after-statement -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wshadow=compatible-local -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-format-truncation -Wno-stringop-truncation -O2 isolationtester.o specparse.o specscanner.o -L../../../src/interfaces/libpq -lpq -L../../../src/port -L../../../src/common   -Wl,--as-needed -Wl,-rpath,'/home/user/.pgenv/pgsql-17.0/lib',--enable-new-dtags  -lpgcommon -lpgport -lz -lreadline -lm  -o isolationtester
make[2]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/test/isolation'
/usr/bin/make -C test/perl all
make[2]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/src/test/perl'
make[2]: Nothing to be done for 'all'.
make[2]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src/test/perl'
make[1]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/src'
```

This is how it stops for me. I don't see an error

**Comment by theory** [comment]
Hrm. so `$HOME/.pgenv/pgsql-17.0` exists?

**Comment by sathishmanohar** [comment]
No.

```
ls .pgenv                                                                                                                                                                                                    
bin   config   patch   pgsql   pgsql-11.10   pgsql-16.4   src   LICENSE.md   README.md
```

```
ls .pgenv/src/                                                                   
postgresql-11.10   postgresql-16.4   postgresql-17.0   postgresql-11.10.tar.bz2   postgresql-16.4.tar.bz2   postgresql-17.0.tar.bz2
```

**Comment by theory** [comment]
Then the install did not finish; no idea why there are no errors. Can you try the patch in #80?

**Comment by sathishmanohar** [comment]
Replaced ~/.pgenv/bin/pgenv with the pgenv from the branch you referred to. Now the psql 17.0 is built successfully.

```
/usr/bin/make -C vacuumlo install
make[2]: Entering directory '/home/user/.pgenv/src/postgresql-17.0/contrib/vacuumlo'
/usr/bin/mkdir -p '/home/user/.pgenv/pgsql-17.0/bin'
/usr/bin/install -c  vacuumlo '/home/user/.pgenv/pgsql-17.0/bin'
make[2]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/contrib/vacuumlo'
make[1]: Leaving directory '/home/user/.pgenv/src/postgresql-17.0/contrib'
pgenv configuration written to file /home/user/.pgenv/config/17.0.conf
PostgreSQL 17.0 built
```

**Comment by theory** [comment]
Yeah, it was the documentation issue, then. Closing as a dupe of #79, thanks!

**Comment by theory** [comment]
FYI: #80 has been merged to fix this issue, and related in v1.3.8.

**Comment by fluca1978** [comment]
I've marked `v1.3.8` as `latest`, since github was reporting `1.3.7` as the latest to download.
